src/genesys/data.py

```diff
@@ -43,7 +43,9 @@ class DataLoaderGenesys:
     Each dataset that is pass must have a "train" split and the content must be a list of dict with at least a "problem" and a "ground_truth" key.
     """
 
-    def __init__(self, config: DataConfig, tokenizer: AutoTokenizer):
+    def __init__(
+        self, config: DataConfig, tokenizer: AutoTokenizer, prime_metric: PrimeMetric, do_tokenization: bool = False
```

Comment:
should we just do Tokenization = false by default ?

Reply:
eh that is the default?

Reply:
sorry I mean, should we just remove this param and just always do =false

Reply:
Oh but we need the result to be tokenized for toploc. Because encode and decode are not inverses. If we do toploc using the text and we tokenize them, we might get a different token sequence and thus get a false negative.

Reply:
So now the code returns the detokenized result (bypass detokenizer in sglang). We save the `output_ids` and the `llm_response` in the json. The `llm_response` being the `tokenizer.decode(output_ids)`

Reply:
My point is that since we pass `do_tokenization=True` I don't see when we use the code to do `do_tokenization=False`

Reply:
Well if we are pretty sure we also want to tokenized output then sure